tracing-flame/README.md

```diff
@@ -0,0 +1,128 @@
+# tracing-flame
+
+A Tracing [Layer][`FlameLayer`] for generating a folded stack trace for generating flamegraphs
+and flamecharts with [`inferno`]
+
+[![Crates.io][crates-badge]][crates-url]
+[![Documentation][docs-badge]][docs-url]
+[![Documentation (master)][docs-master-badge]][docs-master-url]
+[![MIT licensed][mit-badge]][mit-url]
+[![Build Status][actions-badge]][actions-url]
+[![Discord chat][discord-badge]][discord-url]
+![maintenance status][maint-badge]
+
+[Documentation][docs-url] | [Chat][discord-url]
+
+# Overview
+
+[`tracing`] is a framework for instrumenting Rust programs to collect
+scoped, structured, and async-aware diagnostics. `tracing-flame` provides helpers
+for consuming `tracing` instrumentation that can later be visualized as a
+flamegraph/flamechart. Flamegraphs/flamecharts are useful for identifying performance
+issues bottlenecks in an application. For more details, see Brendan Gregg's [post]
+on flamegraphs.
+
+[post]: http://www.brendangregg.com/flamegraphs.html
+
+## Usage
+
+This crate is meant to be used in a two step process:
+
+1. Capture textual representation of the spans that are entered and exited
+   with [`FlameLayer`].
+2. Feed the textual representation into `inferno-flamegraph` to generate the
+   flamegraph or flamechart.
+
+*Note*: when using a buffered writer as the writer for a `FlameLayer`, it is necessary to
+ensure that the buffer has been flushed before the data is passed into
+[`inferno-flamegraph`]. For more details on how to flush the internal writer
+of the `FlameLayer`, see the docs for [`FlushGuard`].
+
+## Layer Setup
+
+```rust
+use std::{fs::File, io::BufWriter};
+use tracing_flame::FlameLayer;
+use tracing_subscriber::{registry::Registry, prelude::*, fmt};
+
+fn setup_global_subscriber() -> impl Drop {
+    let fmt_layer = fmt::Layer::default();
+
+    let (flame_layer, _guard) = FlameLayer::with_file("./tracing.folded").unwrap();
+
+    let subscriber = Registry::default()
+        .with(fmt_layer)
+        .with(flame_layer);
+
+    tracing::subscriber::set_global_default(subscriber).expect("Could not set global default");
```

Comment:
nit/tioli: since we have the `tracing_subscriber` prelude in scope, maybe we should just do
```suggestion
    tracing_subscriber::registry()
        .with(fmt_layer)
        .with(flame_layer)
        .init().
```